api.go

```diff
@@ -154,19 +154,26 @@ func GrabBuiltStaticFiles(dockerImageID, projectName, transferLocation string) {
 	logging.LogToFile(string(res))
 }
 
-func Build(buildDir, projectName, projectPath string) string {
+func Build(buildDir string, dockerInfo DockerInfo) string {
 	c1 := make(chan string)
-	go BuildDockerContainer(c1)
+	quit := make(chan string)
+	go BuildDockerContainer(c1, quit)
+	makePutRequest(dockerInfo, "BLD")
 
 	// Looping until we get notification on the channel c1 that the build has finished
 	for {
 		select {
+		case <-quit:
+			logging.LogToFile("There was an error building the docker container")
+			makePutRequest(dockerInfo, "FAL")
+			return "fail"
 		case buildTag := <-c1:
 			logging.LogToFile("Container built...transfering built files...")
-			GrabBuiltStaticFiles(buildTag, projectName, buildDir)
+			GrabBuiltStaticFiles(buildTag, dockerInfo.REPO_NAME, buildDir)
 			if Config.ENV != "test" {
-				rsyncProject(buildDir+"/public/*", Config.DEPLOYROOTPATH+projectPath)
+				rsyncProject(buildDir+"/public/*", dockerInfo.PATH)
```

Comment:
this line was added recently because we have to rsync to a specific remote folder. Lets discuss.